src/lib.rs

```diff
@@ -238,7 +239,13 @@ impl<'a> ContextExt for Context<'a> {
 #[repr(C)]
 struct FfiWaker {
     vtable: &'static FfiWakerVTable,
-    // Opaque fields after the end of struct.
+    waker: WakerUnion,
+}
+
+#[repr(C)]
+union WakerUnion {
```

Comment:
We could add a variant `_unknow: ()` with comment to mark (also for human) that these two variants are not the only possible cases, since the `FfiWaker` from FFI may contains objects with different layout/size than what we've written here. It is still safe only because `FfiContext::with_context` doesn't actually read this union.